library/NetworkControl.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import logging
+import ipaddress
+
+from library.common import *
+from urllib.parse import quote
+
+
+class NetworkControl(object):
+    def __init__(self, requester, machine_id, interface_name='eth1'):
+        self.log = logging.getLogger(__name__)
+        self.requester = requester
+        self.machine_id = machine_id
+        self.interface_name = interface_name
+        self.machine_ip = ''
+        try:
+            self.interface = self.get_interface()
+        except Exception as err:
+            raise err
+
+    def get_interface(self):
+        url = "/api/v1/machine/{}/interface".format(self.machine_id)
+        try:
+            interfaces = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        for i in interfaces:
+            if str(i['identifier']) == self.interface_name:
+                return i
+
+    def get_ip(self):
```

Comment:
Most of these getters are useless if they're just reaching out a dictionary with slightly different names. They also obscure the MrP names and make it hard for people to link the real sources when debugging.

A better way would be to add "network" = "ip+mask" to the dictionary and get_all return the actual dictionary.

Reply:
you didn't get rid of the getters...